src/utils/conversion.ts

```diff
@@ -4712,3 +4716,95 @@ export function toCustomClaimTypeWithIdentity(
+  portfolioMovementsResult: ApolloQueryResult<Ensured<Query, 'portfolioMovements'>>,
+  filter: string,
+  context: Context
+): HistoricSettlement[] {
+  const data: HistoricSettlement[] = [];
+
+  const getDirection = (fromId: string, toId: string): SettlementDirectionEnum => {
+    const [fromDid] = fromId.split('/');
+    const [toDid] = toId.split('/');
+    const [filterDid, filterPortfolioId] = filter.split('/');
+
+    if (fromId === toId) {
+      return SettlementDirectionEnum.None;
+    }
+
+    if (filterPortfolioId && fromId === filter) {
+      return SettlementDirectionEnum.Outgoing;
+    }
+
+    if (filterPortfolioId && toId === filter) {
+      return SettlementDirectionEnum.Incoming;
+    }
+
+    if (fromDid === filterDid && toDid !== filterDid) {
+      return SettlementDirectionEnum.Incoming;
+    }
+
+    if (toDid === filterDid && fromDid !== filterDid) {
+      return SettlementDirectionEnum.Outgoing;
+    }
+
+    return SettlementDirectionEnum.None;
+  };
+
+  /* eslint-disable @typescript-eslint/no-non-null-assertion */
+  settlementsResult.data.legs.nodes.forEach(({ settlement }) => {
+    const {
+      createdBlock,
+      result: settlementResult,
+      legs: { nodes: legs },
+    } = settlement!;
+
+    const { blockId, hash } = createdBlock!;
+
+    data.push({
+      blockNumber: new BigNumber(blockId),
+      blockHash: hash,
+      status: settlementResult as unknown as SettlementResultEnum,
+      accounts: legs[0].addresses.map(
+        (accountAddress: string) =>
+          new Account({ address: keyToAddress(accountAddress, context) }, context)
+      ),
+      legs: legs.map(({ from, to, fromId, toId, assetId, amount }) => ({
+        asset: new FungibleAsset({ ticker: assetId }, context),
+        amount: new BigNumber(amount).shiftedBy(-6),
+        direction: getDirection(fromId, toId),
+        from: middlewarePortfolioToPortfolio(from!, context),
+        to: middlewarePortfolioToPortfolio(to!, context),
+      })),
+    });
+  });
+
+  portfolioMovementsResult.data.portfolioMovements.nodes.forEach(
+    ({ createdBlock, from, to, fromId, toId, assetId, amount, address: accountAddress }) => {
+      const { blockId, hash } = createdBlock!;
+      data.push({
+        blockNumber: new BigNumber(blockId),
+        blockHash: hash,
+        status: SettlementResultEnum.Executed,
+        accounts: [new Account({ address: keyToAddress(accountAddress, context) }, context)],
+        legs: [
+          {
+            asset: new FungibleAsset({ ticker: assetId }, context),
+            amount: new BigNumber(amount).shiftedBy(-6),
+            direction: getDirection(fromId, toId),
+            from: middlewarePortfolioToPortfolio(from!, context),
+            to: middlewarePortfolioToPortfolio(to!, context),
+          },
+        ],
+      });
+    }
+  );
+  /* eslint-enable @typescript-eslint/no-non-null-assertion */
+
+  return data.sort((a, b) => a.blockNumber.minus(b.blockNumber).toNumber());
```

Comment:
(nit) I was thinking most recent first, since our returned order will be the default order for consumers.

Sure people can sort themselves, but I think block number DESC is most common for a default.

```ts
[1, 2, 3].sort((a, b) => a - b) // 1,2,3
[1, 2, 3].sort((a, b) => b - a) // 3,2,1
```

FWIW I think ASC order makes more sense then clumping settlements and portfolio movements together though like it was before.